vnc/vnc.c

```diff
@@ -1054,40 +866,40 @@ skip_encoding(struct vnc *v, int x, int y, int cx, int cy,
 
     switch (encoding)
     {
-        case ENC_RAW:
+        case RFB_ENC_RAW:
         {
             int need_size = cx * cy * get_bytes_per_pixel(v->server_bpp);
-            LOG(LOG_LEVEL_DEBUG, "Skipping ENC_RAW encoding");
+            LOG(LOG_LEVEL_DEBUG, "Skipping RFB_ENC_RAW encoding");
```

Comment:
Are all of the encoding that are skipped just not supported by xrdp with libvnc? Should the user get a warning message about unsupported encodings?

Reply:
The `skip_encoding()` function is used when we're trying to negotiate a desktop size with the user. We send an update request for a single pixel bit of the screen. If the server responds with an ExtendedDesktopSize rectangle we know the server supports resizing. If it doesn't, we know it doesn't support resizing. In both instances we're not interested in the rest of the data from the server at that time because we're going to either resize the server or ask for a full frame update. `skip_encoding()` is used to pass over data in the stream which we're expecting but which we don't care about.

The VNC protocol prevents the server from sending encodings the client hasn't said it supports. If it does, that's an error.